javagen/src/main/java/com/azure/autorest/template/example/ClientMethodExampleWriter.java

```diff
@@ -64,6 +78,130 @@ public ClientMethodExampleWriter(ClientMethod method, String clientVarName, Prox
+                        });
+                        // assert JSON of first item, or assert count=0
+                        if (ContentType.APPLICATION_JSON.equals(method.getProxyMethod().getRequestContentType())
+                                && responseType.getTypeArguments().length > 0
+                                && ClientModelUtil.isClientModel(responseType.getTypeArguments()[0])
+                                && method.getMethodPageDetails() != null
+                                && response.getBody() instanceof Map) {
+                            Map<String, Object> bodyMap = (Map<String, Object>) response.getBody();
+                            if (bodyMap.containsKey(method.getMethodPageDetails().getSerializedItemName())) {
+                                Object items = bodyMap.get(method.getMethodPageDetails().getSerializedItemName());
+                                if (items instanceof List) {
+                                    List<Object> itemArray = (List<Object>) items;
+                                    if (itemArray.isEmpty()) {
+                                        methodBlock.line("Assertions.assertEquals(0, response.stream().count());");
+                                    } else {
+                                        Object firstItem = itemArray.iterator().next();
+                                        methodBlock.line("%s firstItem = %s;", responseType.getTypeArguments()[0], "response.iterator().next()");
+                                        writeModelAssertion(methodBlock, nodeVisitor, (ClassType) responseType.getTypeArguments()[0], (Map<String, Object>) firstItem, "firstItem");
+                                    }
+                                }
+                            }
+                        }
+                    }
+                } else if (ClientModelUtil.isClientModel(returnType) && response.getBody() instanceof Map) {
+                    // Client Model
+                    // TODO(xiaofei) when return type is boolean (exists through HEAD), or Map/List and primitive types(are they possible?)
+                    writeModelAssertion(methodBlock, nodeVisitor, (ClassType) returnType, (Map<String, Object>) response.getBody(), "response");
+                }
+            } else {
+                methodBlock.line("Assertions.assertNotNull(response);");
+            }
+        };
+    }
+
+    private void addClientModelImports(IType type) {
+        ClientModel clientModel = null;
+        if (type instanceof GenericType) {
+            GenericType responseType = (GenericType) type;
+            if (PagedIterable.class.getSimpleName().equals(responseType.getName())) {
+                if (ClientModelUtil.isClientModel(responseType.getTypeArguments()[0])) {
```

Comment:
For `SyncPoller`, it may need arg 0 and 1?

Reply:
Strange, I didn't add it in codegen and generated code already included those two imports..
Will look into that a little.

Reply:
Not sure if somewhere else "import" the client method. Client method "import" likely contains all parameter model and response model.

Reply:
It's from return type's imports. When adding return type's import(`SyncPoller`), its type arguments are added to imports as well(though not its property types). https://github.com/Azure/autorest.java/blob/f3b79782dec73916f16bd72f8468c563e3184cd3/javagen/src/main/java/com/azure/autorest/model/clientmodel/GenericType.java#L171-L176

It should be fine, as we won't do assertions on properties of those types.